Assets/World/AStarMap.gd

```diff
@@ -50,42 +44,33 @@ func _ready() -> void:
 						if v2 == Vector2i.ZERO:
 							continue
 
+						#prints("v2:", v2, "tile:", tile)
 						if v2_to_index(v2 + tile) in all_points:
-							var ind1 = all_points[v2_to_index(tile)]
-							var ind2 = all_points[v2_to_index(tile + v2)]
-							if !as_node.are_points_connected(ind1, ind2):
-								as_node.connect_points(ind1, ind2, true)
+							var i1 = all_points[v2_to_index(tile)]
+							var i2 = all_points[v2_to_index(tile + v2)]
+							if !as_node.are_points_connected(i1, i2):
+								as_node.connect_points(i1, i2, true)
 
 func _get_tile_item_name(tile: Vector2i) -> String:
-	var tile_item_index = grid_map.get_tile_item(tile)
-	return grid_map.get_item_name(tile_item_index)
+	var tile_item_index = tile_map.get_tile_item(tile)
+	return tile_map.get_item_name(tile_item_index)
 
 func v2_to_index(v2: Vector2i) -> String:
-	return str(int(v2.x)) + "," + str(int(v2.y))
+	#print("v2_to_index(", v2, ")")
+	return str(v2.x) + "," + str(v2.y)
 
-func get_tilemap_path(start: Vector2i, end: Vector2i) -> PackedVector2Array:
+func get_tile_map_path(start: Vector2i, end: Vector2i) -> PackedVector2Array:
 	#print_debug(start, end)
-	var gm_start := grid_map.world_to_tilemap(start)
-	var gm_end := grid_map.world_to_tilemap(end)
+	var tile_map_start: Vector2i = tile_map.local_to_tile_map(start)
+	var tile_map_end: Vector2i = tile_map.local_to_tile_map(end)
 	var start_id := 0
 	var end_id := 0
-	if gm_start in all_points:
-		start_id = all_points[gm_start]
+	if tile_map_start in all_points:
+		start_id = all_points[tile_map_start]
 	else:
 		start_id = as_node.get_closest_point(start)
-	if gm_end in all_points:
-		end_id = all_points[gm_end]
+	if tile_map_end in all_points:
+		end_id = all_points[tile_map_end]
```

Comment:
It will better if it is not this func to verify if tile_map_start is in all_points or not.

Alors instead of dictionary with string key, you can hash the vector like that
int result = vec.y * map.width + vec.x
This func can verify also if vec is in range of the map, so you do not need to verify later.
Plus is more efficient to hash like that and use an array of int instead of a dictionary to search in.